html/modules/custom/reliefweb_fields/components/reliefweb-options/reliefweb-options.js

```diff
@@ -0,0 +1,228 @@
+
+          var terms = [];
+          var items = element.querySelectorAll('[data-term-description]');
+          for (var j = 0, lj = items.length; j < lj; j++) {
+            var item = items[j];
+            var labelElement = item.querySelector('label');
+            terms.push({
+              label: trim(labelElement.textContent || labelElement.innerText),
+              description: item.getAttribute('data-term-description'),
+              for: labelElement.getAttribute('for')
+            })
+          }
+
+          wrappers[fieldName] = {
+            element: element,
+            label: label,
+            terms: terms
+          };
+        }
+        return wrappers;
+      }
+
+      // Remove term description elements from the form.
+      function cleanTermDescriptions(form, popupOnly) {
+        var selector = '.with-term-description';
+        if (!popupOnly) {
+          selector += ', [data-with-term-description]';
+        }
+        var elements = form.querySelectorAll(selector);
+        for (var i = elements.length - 1; i >= 0; i--) {
+          var element = elements[i];
+          element.parentNode.removeChild(element);
+        }
+      }
+
+      // Hide the popup.
+      function hideTermDescriptions(form) {
+        cleanTermDescriptions(form, true);
+      }
+
+      // Show the popup.
+      function showTermDescriptions(form, data) {
+        hideTermDescriptions(form);
+
+        var close = document.createElement('button');
+        close.setAttribute('type', 'button');
+        close.setAttribute('value', 'close');
+        close.appendChild(document.createTextNode(t('Close')));
+        close.addEventListener('click', function (event) {
+          hideTermDescriptions(form);
+        });
+
+        form.addEventListener('keydown', function(event) {
+          const key = event.key;
+          if (key === "Escape") {
+            hideTermDescriptions(form);
+          }
+        });
+
+        var heading = document.createElement('h3');
+        heading.innerText = data.label;
+
+        var content = document.createElement('div');
+        content.className = 'content';
+
+        var dl = document.createElement('dl');
+        for (var i = 0, l = data.terms.length; i < l; i++) {
+          var term = data.terms[i];
+
+          var source = document.getElementById(term.for);
+          var checkbox = document.createElement('input');
+          checkbox.setAttribute('type', source.getAttribute('type'));
+          if (source.checked) {
+            checkbox.setAttribute('checked', 'checked');
+          }
+          checkbox.setAttribute('name', 'shadow-' + source.getAttribute('name'));
+          checkbox.setAttribute('id', 'shadow-' + term.for);
+          checkbox.setAttribute('data-for', term.for);
+          checkbox.addEventListener('change', function (e) {
+            let real = document.getElementById(this.getAttribute('data-for'));
+            real.checked = this.checked;
+          });
```

Comment:
The problem with that approach is that it doesn't respect the maximum of allowed terms. For example by selecting the professional function terms in the popup, I could exceed the limit:

<img width="901" alt="Screen Shot 2022-06-30 at 15 50 06" src="https://user-images.githubusercontent.com/696348/176611492-9aba9210-4eb9-4e89-a729-1060d77d873e.png">

That's also going to cause issues with the `career categories` for jobs where the `theme` options become disabled for some career categories (for example themes are not allowed when selecting "Administration/Finance"):

<img width="442" alt="Screen Shot 2022-06-30 at 15 53 21" src="https://user-images.githubusercontent.com/696348/176612017-181378c6-eb60-4f87-b9b3-dac13bd55030.png">

I haven't tested but maybe another way to handle that could be to transform the original field wrapper into the popup. By that I mean, adding a class to the fieldset wrapper that would result in it being styled as a popup and showing the description via the a `::after` for example (a bit like the option 2 in the RW-601 ticket but with the field styled as a popup). If that works, that could reduce the complexity of the javascript as well.

Reply:
Fixed, attributes are copied and `Drupal.attachBehaviors(popup)` is called